### ISSUE TITLE
make !reconnect have the bot leave+reconnect, instead of just reconnecting

### ISSUE THREAD

**Issue by qixils** [HTSTEM/MusicBot#31]
basically turn !reconnect's current functionality into what you would get from doing **!disconnect**, followed by **!reconnect**

**Comment by Bottersnike** [comment]
That's what it already does? No?

**Comment by Tantusar** [comment]
No, that is not what it already does. It just says, "Already connected, sucks to be you." Okay, it's a little more polite.

**Comment by Bottersnike** [comment]
I'm pretty sure I patched that.

**Comment by Bottersnike** [comment]
For context: https://github.com/HTSTEM/MusicBot/blob/acf88a8b266ffc6690ae2aad186cb476aaace685/musicbot/cogs/music.py#L849 it locates any voice channel it's connected to and force-disconnects.

**Comment by Bottersnike** [comment]
Ensured fixed with https://github.com/HTSTEM/MusicBot/commit/145621957d7f68577c850590f41c8a3f22aa7a16 Please reopen if continuing.

**Comment by qixils** [comment]
![image](https://user-images.githubusercontent.com/13265322/38201127-0e6dda06-3665-11e8-8e2b-744b495f8083.png)

**Comment by Bottersnike** [comment]
Right, so, https://github.com/HTSTEM/MusicBot/blob/acf88a8b266ffc6690ae2aad186cb476aaace685/musicbot/cogs/music.py#L852 ensures that if it has a reference to the voice channel anywhere, it disconnects. The problem would appear to be that for some reason it no longer has a ref.

**Comment by qixils** [comment]
![image](https://user-images.githubusercontent.com/13265322/43615015-c5c2e614-9683-11e8-89e9-2552c75f2544.png)
